auth-web/src/components/auth/common/PaymentMethods.vue

```diff
@@ -333,29 +359,61 @@ export default defineComponent({
-    const paymentsPerAccountType = ConfigHelper.paymentsAllowedPerAccountType()
-
-    const allowedPaymentMethods = computed(() => {
-      const paymentMethods = []
-      if (props.currentOrgType) {
-        const paymentTypes = paymentsPerAccountType[props.currentOrgType]
-        paymentTypes?.forEach((paymentType) => {
-          if (PAYMENT_METHODS[paymentType]) {
-            paymentMethods.push(PAYMENT_METHODS[paymentType])
-          }
+    const paymentMethodSupportedForProducts = computed(() => {
+      const productPaymentMethods = orgStore.productPaymentMethods
+      const { productList } = storeToRefs(useOrgStore())
+      const derivedProductList = props.isCreateAccount ? currentSelectedProducts.value : productList.value
+        .filter(item => item.subscriptionStatus === 'ACTIVE')
+        .map(item => item.code)
+      const paymentMethodProducts = {}
+      for (const [product, methods] of Object.entries(productPaymentMethods)) {
+        methods.forEach(method => {
+          paymentMethodProducts[method] ??= []
+          paymentMethodProducts[method].push(product)
         })
       }
-      if (currentOrgPaymentDetails?.eftEnable) {
-        paymentMethods.push(PAYMENT_METHODS[PaymentTypes.EFT])
+
+      const paymentMethodSupported = {}
+      Object.entries(paymentMethodProducts).forEach(([key, values]) => {
+        paymentMethodSupported[key] = derivedProductList.every(subscription => (values as Array<string>).includes(subscription))
+      })
+      paymentMethodSupported[PaymentTypes.CREDIT_CARD] = paymentMethodSupported[PaymentTypes.DIRECT_PAY]
+      return paymentMethodSupported
+    })
+
+    const filteredPaymentMethods = computed(() => {
+      if (!props.isEditing && !props.isCreateAccount && state.selectedPaymentMethod) {
+        return [PAYMENT_METHODS[state.selectedPaymentMethod]]
       }
+      const methodSupportPerProduct = paymentMethodSupportedForProducts.value
+      const paymentMethods = []
+      const isGovmOrg = props.currentOrgType === AccessType.GOVM
+      const paymentTypes = isGovmOrg ? [PaymentTypes.EJV] : [PaymentTypes.PAD, PaymentTypes.CREDIT_CARD,
+        PaymentTypes.ONLINE_BANKING, PaymentTypes.BCOL, PaymentTypes.EFT]
+      paymentTypes.forEach((paymentType) => {
+        if (paymentType === PaymentTypes.EFT && !currentOrgPaymentDetails?.eftEnable) {
+          return
+        }
+        const paymentMethod = PAYMENT_METHODS[paymentType]
+        if (paymentMethod) {
+          paymentMethod.supported = methodSupportPerProduct[paymentType]
+          if (props.isCreateAccount) {
+            if (currentSelectedProducts.value.length === 0) {
+              paymentMethod.supported = false
+            }
+            if (paymentType === state.selectedPaymentMethod && !paymentMethod.supported) {
+              state.selectedPaymentMethod = ''
+              emit('payment-method-selected', state.selectedPaymentMethod)
```

Comment:
This is not so good, side effects can happen due to this, I'll figure out a better way to do this